packages/common-all/src/util/treeUtil.ts

```diff
@@ -39,25 +42,119 @@ export type TreeNode = {
 
 export class TreeUtils {
   static generateTreeData(
-    allNotes: NotePropsByIdDict,
-    domains: NoteProps[]
+    noteDict: NotePropsByIdDict,
+    sidebars: Sidebars
   ): TreeMenu {
-    // --- Calc
-    const roots = domains
-      .map((note) => {
-        return TreeUtils.note2Tree({
-          noteId: note.id,
-          noteDict: allNotes,
-        });
-      })
-      .filter((ent): ent is TreeMenuNode => !_.isUndefined(ent));
-
-    const child2parent: { [key: string]: string | null } = {};
-    Object.entries(allNotes).forEach(([noteId, note]) => {
-      child2parent[noteId] = note.parent;
+    function itemToNoteId(item: SidebarItem) {
+      const { type } = item;
+      switch (type) {
+        case "category": {
+          return item.link?.id;
+        }
+        case "note": {
+          return item.id;
+        }
+        default:
+          assertUnreachable(type);
+      }
+    }
+
+    function itemToTreeMenuNode(
+      sidebarItem: SidebarItem,
+      opts: {
+        child2parent: Record<string, string | null>;
+        parent: string | null;
+        notesLabelById: Record<string, string>;
+      }
+    ): TreeMenuNode | undefined {
+      const { child2parent, parent, notesLabelById } = opts;
+
+      const noteId = itemToNoteId(sidebarItem);
+      const note = noteDict[noteId] as NoteProps | undefined; // explicitly casting since `noUncheckedIndexedAccess` is currently not enabled
+
+      if (_.isUndefined(note)) {
+        return undefined;
+      }
+
+      const icon = do_(() => {
+        if (note.schema) {
+          return TreeMenuNodeIcon.bookOutlined;
+        } else if (note.fname.toLowerCase() === TAGS_HIERARCHY_BASE) {
+          return TreeMenuNodeIcon.numberOutlined;
+        } else if (note.stub) {
+          return TreeMenuNodeIcon.plusOutlined;
+        }
+        return null;
+      });
+
+      const title = sidebarItem.label ?? note.title;
+
+      notesLabelById[note.id] = title;
+
+      const fm = PublishUtils.getPublishFM(note);
+
+      const treeMenuNode: TreeMenuNode = {
+        key: note.id,
+        title,
+        icon,
+        hasTitleNumberOutlined: note.fname.startsWith(TAGS_HIERARCHY),
+        vaultName: VaultUtils.getName(note.vault),
+        navExclude: fm.nav_exclude || false,
+        children: [],
+      };
+
+      if (child2parent[note.id] === undefined) {
+        child2parent[note.id] = parent;
+      }
+
+      if (sidebarItem.type === "category") {
+        treeMenuNode.children = sidebarItem.items
+          .map((item) =>
+            itemToTreeMenuNode(item, {
+              child2parent,
+              parent: note.id,
+              notesLabelById,
+            })
+          )
+          .filter((maybeTreeMenuNode): maybeTreeMenuNode is TreeMenuNode =>
+            Boolean(maybeTreeMenuNode)
+          );
+      }
+
+      return treeMenuNode;
+    }
+
+    const treeMenuMap = _.mapValues(sidebars, (sidebar) => {
```

Comment:
we should not be supporting multiple sidebars for v1

Reply:
I left room for the option of supporting them in the future since its just a simple layer on existing code and thought adding it later would be a breaking change inside the structure of the sidebar config. But I may have overthought there. I feel its good to remove that now and made it only support single sidebars. [`67e554d` (#3448)](https://github.com/dendronhq/dendron/pull/3448/commits/67e554dd7fd40c71103560544301bbc46feba3cf)